content/2021-06-11-make-your-life-easier.md

```diff
@@ -0,0 +1,156 @@
++++
+title = "Make your life easier"
+template = "page.html"
+date = 2021-06-11T22:00:00Z
+
+[taxonomies]
+tags = ["Tech"]
++++
+
+If you write software for long enough you begin to develop instincts about the
+relative difficulty of different problems and some of those solidify into
+rules of thumb. Like all such things they are not really _rules_ but patterns
+of thinking that you can choose to deviate from when you have good reason to
+do so. The following are some that I have come across recently. I can't claim
+that any of them are original ideas, but they are some of the things that
+guide my work on a daily basis.
+
+# Choose simpler graph data structures
+
+Many problems are graph problems of one sort or another and there's a
+hierarchy of increasing complexity of graph types:
+
+- list
+- tree
+- acyclic graph
+- cyclic graph
+
+All of those are technically graphs, but the amount of work you need to do to
```

Comment:
It sounds like a simple recommendation here is: Add limits. You can always make customers pay more to raise limits in the future.